diff --git a/qtbase/src/3rdparty/libpng/pngpriv.h b/qtbase/src/3rdparty/libpng/pngpriv.h
index 8211bd43..789d8124 100644
--- a/qtbase/src/3rdparty/libpng/pngpriv.h
+++ b/qtbase/src/3rdparty/libpng/pngpriv.h
@@ -520,7 +520,8 @@

 #  if (defined(__MWERKS__) && defined(macintosh)) || defined(applec) || \
     defined(THINK_C) || defined(__SC__) || defined(TARGET_OS_MAC)
-   /* We need to check that <math.h> hasn't already been included earlier
+#    if !defined(__APPLE__)
+   /* We need to check that <math.h> hasn't already been included earlier
     * as it seems it doesn't agree with <fp.h>, yet we should really use
     * <fp.h> if possible.
     */
@@ -528,6 +529,9 @@
 #      include <fp.h>
 #    endif
 #  else
+#      include <math.h>
+#    endif
+#  else
 #    include <math.h>
 #  endif
 #  if defined(_AMIGA) && defined(__SASC) && defined(_M68881)
